library/index.c

```diff
@@ -323,24 +323,25 @@ int eblob_index_blocks_fill(struct eblob_base_ctl *bctl)
 }
 
 
-static struct eblob_disk_control *eblob_find_on_disk(struct eblob_backend *b,
-		struct eblob_base_ctl *bctl, struct eblob_disk_control *dc,
+static int eblob_find_on_disk(struct eblob_backend *b,
```

Comment:
No, it doesn't. Please read what I wrote again or check the code exactly 3 lines below what you have shown.

There is a range index tree which splits index file into ranges of 40 keys by default. These ranges are being looked for in eblob_index_blocks_search_nolock() function. It returns pointer to the control structure, which contains start and end keys for range of keys which covers requested key. Code then checks mapped area within those start/end keys and runs bsearch(). It is possible that there are several the same keys in the index file, thus there is linear search right after bsearch().

You broke that logic basically by not taking into account range index tree. I introduced this tree years ago exactly to solve  problem with multiple random reads over the whole index file. Now it is replaced with single jump into the index file and limit its search by the size of the index range (40 entries by default).